bpfctl/src/main.rs

```diff
@@ -234,6 +269,115 @@ async fn main() -> anyhow::Result<()> {
             let response = client.load(request).await?.into_inner();
             println!("{}", response.id);
         }
+        Commands::LoadFromImage(l) => {
+            let prog_type = match l.command {
+                LoadCommands::Xdp { .. } => ProgramType::Xdp,
+                LoadCommands::Tc { .. } => ProgramType::Tc,
+                LoadCommands::Tracepoint { .. } => ProgramType::Tracepoint,
+            };
+            let attach_type = match &l.command {
+                LoadCommands::Xdp {
+                    iface,
+                    priority,
+                    proceed_on,
+                } => {
+                    let mut proc_on = Vec::new();
+                    if !proceed_on.is_empty() {
+                        for i in proceed_on.iter() {
+                            let action = ProceedOn::try_from(i.to_string())?;
```

Comment:
I looked around clap for a bit to see if there was higher level tooling to parse input and convert the string values to enums, e.t.c. but nothing stood out right away. Overall it would seem nice if we could make the parsing of string input into the explicit types and enum variants they correspond with happen in it's own step discreetly in time, but I don't think it needs to hold up this PR just maybe part of a later refactoring opportunity.

(feel free to consider this comment resolved at your discretion)